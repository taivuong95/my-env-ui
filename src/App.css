.container{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100vh;
    background: linear-gradient(180deg
,rgba(50,94,189,.913603),rgba(25,107,138,.779149) 60%,rgba(0,172,255,.319765));
}

.current-weather-layout{
  height: 60%;
  display: flex;
    flex-direction: column;
    justify-content: space-between;
}




.chart-layout{
  position: relative;
  overflow-x: scroll;
  overflow-y: hidden;
  height: 40%;
  background-color: #fff;
}

