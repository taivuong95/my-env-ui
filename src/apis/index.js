
import axios from 'axios';
export const API_KEY = '9cc29ecca0f48e0f260928af174aaa33';

export const getWeatherData = async (location) => {
    const URL = `https://api.openweathermap.org/data/2.5/weather?q=${location}&appid=${API_KEY}&units=metric`
    const request = await axios.get(URL);
    return request;
}