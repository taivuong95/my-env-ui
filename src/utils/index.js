export const KelvinToCelsius = (kelvin) => {
    return kelvin - 273.15;
}