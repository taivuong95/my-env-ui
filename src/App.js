import { useEffect, useState } from 'react';
import "./App.css";
import Navigation from './components/Navigation/Navigation';
import WeatherDetail from './components/WeatherDetail/WeatherDetail';
import WeatherInfo from './components/WeatherInfo/WeatherInfo';
import TideSunChart from './components/TideSunChart/TideSunChart';
import WeatherSearch from './components/WeatherSearch/WeatherSearch';
import { getWeatherData } from './apis';
import Context from './contexts/Context';
import Error from './components/UI/Error'

function App() {
  const [weather, setWeather] = useState();
  const [city, setCity] = useState();
  const [error, setError] = useState();
  const [loading, setLoading] = useState();

  const getWeatherInfo = async (e) => {
    e.preventDefault();
    setLoading(true);
    const location = e.target.elements.location.value;
    if (!location) {
      setLoading(true)
      setWeather(null);
      return setError("Please enter the name of the city.");
    }
    try {
      const request = getWeatherData(location)
      const { data } = await request;
      setWeather(data);
      setCity(data.name);
      setError(null);
      setLoading(false);
    } catch (error) {
      setError('Please enter valid city name.');
      setWeather(null);
      setCity(null);
    }
  }
  useEffect(() => {

  }, [])
  return (
    <div className='container'>
      <Context.Provider value={{ getWeatherInfo, weather, city, loading }}>
        <div className='current-weather-layout'>
          <Navigation />
          <WeatherSearch />
          {weather && <WeatherDetail />}
          {error && <Error error={error} />}
          <WeatherInfo />
        </div>
        <div className='chart-layout'>
          <TideSunChart />
        </div>
      </Context.Provider>
    </div>
  );
}

export default App;
