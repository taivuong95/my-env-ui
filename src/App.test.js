import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr } from "./utils/index";
import App from "./App";
import { mockData } from "./test/mockData";
import { getWeatherData } from "./apis";
import axios from "axios";
jest.mock("axios");
const updateInput = (wrapper, instance, newValue) => {
  const input = wrapper.find(instance);
  input.simulate("change", {
    currentTarget: { value: newValue },
  });
  return wrapper.find(instance);
};
const setup = () => shallow(<App />);
describe("App Component", () => {
  test("renders without error", () => {
    const wrapper = setup();
    const appComponent = findByTestAttr(wrapper, "container");
    expect(appComponent.length).toBe(1);
  });

  test("submits the form", () => {
    // const resp = { mockData };
    // axios.get.mockImplementationOnce(() => Promise.resolve(resp));
    // const wrapper = setup();
    // updateInput(wrapper, '[data-test="city"]', "Paris");

    // wrapper
    //   .find('[data-test="weather-search__form"]')
    //   .simulate("submit", { preventDefault: () => {} });

    // expect(getWeatherData).toHaveBeenCalledWith("Paris");
  });
});
