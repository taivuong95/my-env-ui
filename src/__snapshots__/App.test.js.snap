// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component expect to render App component 1`] = `
<div
  className="container"
  data-test="container"
>
  <ContextProvider
    value={
      Object {
        "city": undefined,
        "loading": undefined,
        "setWeatherInfo": [Function],
        "weather": undefined,
      }
    }
  >
    <div
      className="current-weather-layout"
    >
      <Navigation />
      <WeatherSearch />
      <WeatherInfo />
    </div>
    <div
      className="chart-layout"
    >
      <TideSunChart />
    </div>
  </ContextProvider>
</div>
`;
